app/src/pages/inside/settingsPage/settingTabs/notificationsTab/forms/emailCasesList.jsx

```diff
@@ -0,0 +1,168 @@
+import React, { Component, Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { defineMessages, injectIntl, intlShape } from 'react-intl';
+import classNames from 'classnames/bind';
+import { showModalAction } from 'controllers/modal';
+import { compareObjectsByKeys, uniqueId } from 'common/utils';
+import { GhostButton } from 'components/buttons/ghostButton';
+import { EmailCase } from './emailCase';
+import styles from './forms.scss';
+import PlusIcon from './img/ic-plus-inline.svg';
+import { OWNER, defaultRecipient } from './constants';
+
+const messages = defineMessages({
+  addNewRuleButton: {
+    id: 'NotificationsForm.addNewRuleButton',
+    defaultMessage: 'Add New Rule',
+  },
+  duplicationErrorMessage: {
+    id: 'NotificationsForm.duplicationErrorMessage',
+    defaultMessage: "Such notification rule already exists. You can't create duplicate.",
+  },
+});
+const cx = classNames.bind(styles);
+
+@injectIntl
+@connect(null, {
+  showModal: showModalAction,
+})
+export class EmailCasesList extends Component {
+  static propTypes = {
+    emailCases: PropTypes.array,
+    showModal: PropTypes.func,
+    onSubmit: PropTypes.func,
+    intl: intlShape.isRequired,
+    configurable: PropTypes.bool,
+  };
+  static defaultProps = {
+    emailCases: [],
+    showModal: () => {},
+    onSubmit: () => {},
+    configurable: false,
+  };
+  state = {
+    emailCases: this.props.emailCases.map((emailCase) => {
```

Comment:
Can this functionality be achieved by using redux-form's `FieldArray`?

Reply:
https://redux-form.com/7.3.0/docs/api/fieldarray.md/#-code-component-component-function-code-required-

"A Component or stateless function to render the field array."